### PR TITLE
Convert ps to newshell

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — CHANGES_REQUESTED** [review]
Probably should treat strings as zero-terminated by default:
```
[XX] db/cmd/cmd_wD woE woD rc4
RZ_NOPLUGINS=1 /private/var/folders/bc/yvvw2g2s4jlg9r8x7tf61xv800009d/T/woodpecker-local-693079402/rizinorg/rizin/prefix/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -Qc 'w helloworld
woE rc4 s:key
woD rc4 s:key
ps
' =
-- stdout
--- expected
+++ actual
@@ -1,1 +1,1 @@
-helloworld
+helloworld\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00

-- stderr
WARNING: core: Written 256 byte(s)
WARNING: core: Written 256 byte(s)
```

**Comment by wargio** [comment]
> Probably should treat strings as zero-terminated by default:

Yes i just switch it.

File: librz/core/cmd/cmd_print.c [hunk 1]
```diff
@@ -2564,6 +2547,221 @@ static void core_print_raw_buffer(RzStrStringifyOpt *opt) {
...
+		opt.length = length;
+		opt.encoding = encoding;
+		opt.stop_at_nil = stop_at_nil;
+		core_print_raw_buffer(&opt);
+		break;
+	case RZ_OUTPUT_MODE_JSON:
+		print_json_string(core, buffer, length, encoding, stop_at_nil);
+		break;
+	default:
+		RZ_LOG_ERROR("core: unsupported output mode\n");
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c:2573]: rz_warn_if_reached?

File: librz/core/cmd/cmd_print.c [hunk 2]
```diff
@@ -2564,6 +2547,221 @@ static void core_print_raw_buffer(RzStrStringifyOpt *opt) {
...
+}
+
+RZ_IPI RzCmdStatus rz_print_pascal_string_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	RzStrStringifyOpt opt = { 0 };
+	bool big_endian = rz_config_get_b(core->config, "cfg.bigendian");
+	ut64 string_len = 0;
+	ut32 offset = 0;
+
+	if (!strcmp(argv[1], "8")) {
+		string_len = (ut64)core->block[0];
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c:2695]: Will it work for BE platforms too?

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/cmd/cmd_print.c (hunk 2 above)
Comment (wargio) [reply librz/core/cmd/cmd_print.c:2695]: yes, seems that the format is like that, either first byte, half-word, word, quad-word and then the string.

File: librz/core/cmd/cmd_print.c (hunk 1 above)
Comment (wargio) [reply librz/core/cmd/cmd_print.c:2573]: i think both are fine.

**Review by XVilka — APPROVED** [review]

**Pull request merged**